core/core/test/serverspec/spec/master/master_spec.rb

```diff
@@ -1,28 +1,166 @@
 require 'spec_helper'
+require 'base64'
 
-describe service('kubelet') do
-  it { should be_enabled }
-  it { should be_running }
+
+describe 'Check if kubelet service is up' do
+  describe service('kubelet') do
+    it { should be_enabled }
+    it { should be_running }
+  end
+end
+
+describe 'Check if kube-scheduler is up' do
+  describe process('kube-scheduler') do
+    it { should be_enabled }
+    it { should be_running }
+  end
+end
+
+describe 'Check if kube-scheduler is up - 2nd test' do
+  describe command('ps -C kube-scheduler --no-headers') do
+    its(:stdout) { should match /kube-scheduler/ }
+    its(:exit_status) { should eq 0 }
+  end
 end
 
-describe service('kube-controller') do
+describe 'Check if kube-controller is up' do
+  describe process('kube-controller') do
     it { should be_enabled }
     it { should be_running }
   end
+end
+
+describe 'Check if kube-controller is up - 2nd test' do
+  describe command('ps -C kube-controller --no-headers') do
+    its(:stdout) { should match /kube-controller/ }
+    its(:exit_status) { should eq 0 }
+  end
+end
 
-describe service('kube-apiserver') do
+describe 'Check if kube-apiserver is up' do
+  describe process("kube-apiserver") do
     it { should be_enabled }
     it { should be_running }
   end
+end
+
+describe 'Check if kube-apiserver is up - 2nd test' do
```

Comment:
Please rename test (and maybe other) to match what they actually test. For example this test can have a name:
`Check if kube-apiserver is on process list`